### PR TITLE
File Restructure, ESLint Loosening, List Optimization, API Mockup, Search Filter Commit

### PR DESCRIPTION
**Pull request by gkemp94**
**File Restructure**
Moves component files into component folder as is standard

**Loosens up ESLint** 
Makes some development easier and removes pointless rules

**API Mockup** 
Moved data fetching to function that currently returns fake promise, but will return real promise once we have the back-end done

**Search Filter**
Search filters the items in the list view now

**Changes Scroll View to FlatList**
Huge performance boost for user.

Cheers, 
George

### CODE REVIEW

**Comment by SmilingKat** [comment]
The code seems to look okay but I don't have access to the branch from my PC, therefore I cannot review it. It seems to be in your own personal repository. Please create a branch in this repo and commit so that it can be tested by us.

**Comment by SmilingKat** [comment]
Hi George,

please resolve the conflict as stated above. Once this is done, I'm happy to approve this pull request. 

Thanks,
Katja

File: .eslintrc.json [hunk 1]
```diff
@@ -3,15 +3,19 @@
...
         "object-shorthand": "off",
+        "global-require": "off",
+        "no-param-reassign" : "off",
         "react/jsx-filename-extension": "off",
+        "linebreak-style": "off",
         "no-underscore-dangle": ["error", { "allowAfterThis": true }],
-        "comma-dangle": ["error", "never"],
         "import/prefer-default-export": "off",
-		 "require-jsdoc": ["error", {
+		 "require-jsdoc": ["off", {
```
Comment (lukekearney) [inline .eslintrc.json]: Why are we disabling the requirement for jsdoc?

**Comment by gkemp94** [comment]
@SmilingKat @lukekearney Yeah I apologize, I'm going to cancel this pull request. The final commit (the one you have issues with), wasn't meant to be merged in with this pull request (guess I'm still learning git).

**Pull request closed without merging**